packages/amplify-graphql-relational-transformer/src/resolver/ddb-references-generator.ts

```diff
@@ -165,8 +165,117 @@ export class DDBRelationalReferencesResolverGenerator extends DDBRelationalResol
     ctx.resolvers.addResolver(object.name.value, field.name.value, resolver);
   };
 
-  makeHasOneGetItemConnectionWithKeyResolver = (_config: HasOneDirectiveConfiguration, _ctx: TransformerContextProvider): void => {
-    // TODO: Implement hasOne resolver -- this should be nearly identical to hasMany
+  makeHasOneGetItemConnectionWithKeyResolver = (config: HasOneDirectiveConfiguration, ctx: TransformerContextProvider): void => {
```

Comment:
There's a lot of unnecessary duplicate code introduced here. The intent is to define the working implementation as a point of reference, get it in the feature branch, then follow up with improvements. Tracking this work in the feature branch PR description as:
>  Consolidate resolver implementations for ddb references to reduce duplicate code.